Build failure when elasticsearch feature is enabled

Reported by John in slack. When i execute this command to install zebra:

```
cargo install --features="elasticsearch" --locked --git https://github.com/ZcashFoundation/zebra --tag v1.0.0-rc.8 zebrad
```

I get the following build error:
```
...
   Compiling zebra-state v1.0.0-beta.24 (/home/oxarbitrage/.cargo/git/checkouts/zebra-d0d93d7ded255019/0190882/zebra-state)
error[E0277]: the trait bound `zebra_chain::block::Block: Serialize` is not satisfied
   --> zebra-state/src/service/finalized_state.rs:388:23
    |
388 |                 .push(serde_json::json!(block).to_string());
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^
    |                       |
    |                       the trait `Serialize` is not implemented for `zebra_chain::block::Block`
    |                       required by a bound introduced by this call
    |
    = help: the following other types implement trait `Serialize`:
              &'a T
              &'a mut T
              ()
              (T0, T1)
              (T0, T1, T2)
              (T0, T1, T2, T3)
              (T0, T1, T2, T3, T4)
              (T0, T1, T2, T3, T4, T5)
            and 276 others
    = note: required for `Arc<zebra_chain::block::Block>` to implement `Serialize`
    = note: 2 redundant requirements hidden
    = note: required for `&&Arc<zebra_chain::block::Block>` to implement `Serialize`
note: required by a bound in `to_value`
   --> /home/oxarbitrage/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.96/src/value/mod.rs:942:8
    |
942 |     T: Serialize,
    |        ^^^^^^^^^ required by this bound in `to_value`
    = note: this error originates in the macro `json_internal` which comes from the expansion of the macro `serde_json::json` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `zebra-state` due to previous error
error: failed to compile `zebrad v1.0.0-rc.8 (https://github.com/ZcashFoundation/zebra?tag=v1.0.0-rc.8#01908822)`, intermediate artifacts can be found at `/tmp/cargo-install27LFAQ`
$
```

This is also happening without `--locked` argument.

--- a/zebra-chain/Cargo.toml
+++ b/zebra-chain/Cargo.toml
@@ -22,6 +22,9 @@ getblocktemplate-rpcs = [
     "zcash_address",
 ]
 
+# Experimental elasticsearch support
+elasticsearch = []
+
 # Test-only features
 
 proptest-impl = [

--- a/zebra-chain/src/block.rs
+++ b/zebra-chain/src/block.rs
@@ -43,7 +43,10 @@ pub use arbitrary::LedgerState;
 
 /// A Zcash block, containing a header and a list of transactions.
 #[derive(Clone, Debug, Eq, PartialEq)]
-#[cfg_attr(any(test, feature = "proptest-impl"), derive(Serialize))]
+#[cfg_attr(
+    any(test, feature = "proptest-impl", feature = "elasticsearch"),
+    derive(Serialize)
+)]
 pub struct Block {
     /// The block header, containing block metadata.
     pub header: Arc<Header>,

--- a/zebra-chain/src/transaction.rs
+++ b/zebra-chain/src/transaction.rs
@@ -63,7 +63,10 @@ use crate::{
 /// internally by different enum variants. Because we checkpoint on Canopy
 /// activation, we do not validate any pre-Sapling transaction types.
 #[derive(Clone, Debug, PartialEq, Eq)]
-#[cfg_attr(any(test, feature = "proptest-impl"), derive(Serialize))]
+#[cfg_attr(
+    any(test, feature = "proptest-impl", feature = "elasticsearch"),
+    derive(Serialize)
+)]
 pub enum Transaction {
     /// A fully transparent transaction (`version = 1`).
     V1 {

--- a/zebra-chain/src/transparent.rs
+++ b/zebra-chain/src/transparent.rs
@@ -66,7 +66,10 @@ pub const EXTRA_ZEBRA_COINBASE_DATA: &str = "z\u{1F993}";
 //
 // TODO: rename to ExtraCoinbaseData, because height is also part of the coinbase data?
 #[derive(Clone, Eq, PartialEq)]
-#[cfg_attr(any(test, feature = "proptest-impl"), derive(Serialize))]
+#[cfg_attr(
+    any(test, feature = "proptest-impl", feature = "elasticsearch"),
+    derive(Serialize)
+)]
 pub struct CoinbaseData(
     /// Invariant: this vec, together with the coinbase height, must be less than
     /// 100 bytes. We enforce this by only constructing CoinbaseData fields by
@@ -110,7 +113,11 @@ impl std::fmt::Debug for CoinbaseData {
 ///
 /// A particular transaction output reference.
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
-#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary, Serialize))]
+#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]
+#[cfg_attr(
+    any(test, feature = "proptest-impl", feature = "elasticsearch"),
+    derive(Serialize)
+)]
 pub struct OutPoint {
     /// References the transaction that contains the UTXO being spent.
     ///
@@ -145,7 +152,10 @@ impl OutPoint {
 
 /// A transparent input to a transaction.
 #[derive(Clone, Debug, Eq, PartialEq)]
-#[cfg_attr(any(test, feature = "proptest-impl"), derive(Serialize))]
+#[cfg_attr(
+    any(test, feature = "proptest-impl", feature = "elasticsearch"),
+    derive(Serialize)
+)]
 pub enum Input {
     /// A reference to an output of a previous transaction.
     PrevOut {
@@ -383,9 +393,10 @@ impl Input {
 /// that spends my UTXO and sends 1 ZEC to you and 1 ZEC back to me
 /// (just like receiving change).
 #[derive(Clone, Debug, Eq, PartialEq, Hash)]
+#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary, Deserialize))]
 #[cfg_attr(
-    any(test, feature = "proptest-impl"),
-    derive(Arbitrary, Serialize, Deserialize)
+    any(test, feature = "proptest-impl", feature = "elasticsearch"),
+    derive(Serialize)
 )]
 pub struct Output {
     /// Transaction value.

--- a/zebra-state/Cargo.toml
+++ b/zebra-state/Cargo.toml
@@ -30,6 +30,7 @@ proptest-impl = [
 elasticsearch = [
     "dep:elasticsearch",
     "dep:serde_json",
+    "zebra-chain/elasticsearch",
 ]
 
 [dependencies]

--- a/zebrad/Cargo.toml
+++ b/zebrad/Cargo.toml
@@ -36,6 +36,7 @@ getblocktemplate-rpcs = [
 
 elasticsearch = [
     "zebra-state/elasticsearch",
+    "zebra-chain/elasticsearch",
 ]
 
 sentry = ["dep:sentry"]

